x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -153,6 +154,9 @@ private <T> T invokeParser(
             return result.apply(new AstBuilder(new ExpressionBuilder.ParsingContext(params, metrics)), tree);
         } catch (StackOverflowError e) {
             throw new ParsingException("ESQL statement is too large, causing stack overflow when generating the parsing tree: [{}]", query);
+            // likely thrown by an invalid popMode (such as extra closing parenthesis)
+        } catch (EmptyStackException ese) {
+            throw new ParsingException("Invalid query [{}]", query);
```

Comment:
Unfortunately exception is very generic and does not bring any details on position, 
however I wonder if we could make any assumptions as for what went wrong here? 

It might be frustrating to see message like this for not too complex query like:
```
                stats
                min = min(a),
                max = max(a)) WHERE (a % 3 > 10 OR a / 2 > 100),
                avg = avg(a) WHERE a / 2 > 100
                BY a
```
note `max(a))`.

We know this could be caused by extra `)` or `]`. I wonder if we should attempt to count them in here to highlight this could be an issue?